### PR TITLE
Add JWT support as an option for sign in

### PR DESCRIPTION
**Pull request by andybigbear**
Since we start to support sign in Via JWT. need to change TSC as well for adopt this feature.
this CL:
1. changed the TSC for JWT sign in support
2. update the sample/login as an example
3. create another example of invoking API using JWT

### CODE REVIEW

File: samples/login.py [hunk 1]
```diff
@@ -49,35 +54,69 @@ def sample_define_common_options(parser):
...
     # Make sure we use an updated version of the rest apis, and pass in our cert handling choice
-    server = TSC.Server(args.server, use_server_version=True, http_options={"verify": check_ssl_certificate})
+    server = TSC.Server(args.server, use_server_version=True)
     server.auth.sign_in(tableau_auth)
     print("Logged in successfully")
 
     return server
 
 
+def getSingleCredentialType(args):
```
Comment (jacalata) [inline samples/login.py:92]: all this is unnecessary if we add jwt to the argument group

File: tableauserverclient/models/json_web_token_auth.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class JsonWebTokenAuth(object):
+    def __init__(self, json_web_token, site_id=None):
+        self.json_web_token = json_web_token
```
Comment (jacalata) [inline tableauserverclient/models/json_web_token_auth.py:3]: why pass in a site id if it never gets used?

**Review by jacalata — COMMENTED** [review]
I have an existing implementation of this that adds JWT as a sub-type of Credentials, which I think is a cleaner design that won't lead to a proliferation of sign_in_with_x methods.  How do you feel about using that?

**Comment by andybigbear** [comment]
> 

Hi Jac, if you already have a design for Jwt sign in, I'm up for it as well. Could I review your design as well? If all is good, we can close this CL.

**Comment by jacalata** [comment]
> > 
> 
> Hi Jac, if you already have a design for Jwt sign in, I'm up for it as well. Could I review your design as well? If all is good, we can close this CL.

It's based on this: https://github.com/tableau/server-client-python/pull/1032

**Pull request closed without merging**